server/sonar-server/src/main/java/org/sonar/server/computation/issue/commonrule/CommentDensityRule.java

```diff
@@ -0,0 +1,75 @@
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.issue.commonrule;
+
+import com.google.common.base.Optional;
+import org.sonar.api.measures.CoreMetrics;
+import org.sonar.server.computation.component.Component;
+import org.sonar.server.computation.measure.Measure;
+import org.sonar.server.computation.measure.MeasureRepository;
+import org.sonar.server.computation.metric.Metric;
+import org.sonar.server.computation.metric.MetricRepository;
+import org.sonar.server.computation.qualityprofile.ActiveRule;
+import org.sonar.server.computation.qualityprofile.ActiveRulesHolder;
+import org.sonar.server.rule.CommonRuleKeys;
+
+import static java.lang.String.format;
+
+public class CommentDensityRule extends CommonRule {
+
+  private final MeasureRepository measureRepository;
+  private final Metric commentDensityMetric;
+  private final Metric commentLinesMetric;
+  private final Metric nclocMetric;
+
+  public CommentDensityRule(ActiveRulesHolder activeRulesHolder, MeasureRepository measureRepository, MetricRepository metricRepository) {
+    super(activeRulesHolder, CommonRuleKeys.INSUFFICIENT_COMMENT_DENSITY);
+    this.measureRepository = measureRepository;
+    this.commentDensityMetric = metricRepository.getByKey(CoreMetrics.COMMENT_LINES_DENSITY_KEY);
+    this.commentLinesMetric = metricRepository.getByKey(CoreMetrics.COMMENT_LINES_KEY);
+    this.nclocMetric = metricRepository.getByKey(CoreMetrics.NCLOC_KEY);
+  }
+
+  @Override
+  protected CommonRuleIssue doProcessFile(Component file, ActiveRule activeRule) {
+    if (!file.getFileAttributes().isUnitTest()) {
+      Optional<Measure> commentDensityMeasure = measureRepository.getRawMeasure(file, commentDensityMetric);
+      Optional<Measure> commentLinesMeasure = measureRepository.getRawMeasure(file, commentLinesMetric);
+      Optional<Measure> nclocMeasure = measureRepository.getRawMeasure(file, nclocMetric);
+      // FIXME
+      double minCommentDensity = 25.0;
+      if (commentDensityMeasure.isPresent() && nclocMeasure.isPresent() && nclocMeasure.get().getIntValue() > 0 && commentDensityMeasure.get().getDoubleValue() < minCommentDensity) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 182 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)